src/networking.c

```diff
@@ -3320,16 +3320,14 @@ void flushSlavesOutputBuffers(void) {
  * failover procedure implemented by CLUSTER FAILOVER.
  *
  * The function always succeed, even if there is already a pause in progress.
- * In such a case, the duration is set to the maximum and new end time and the
- * type is set to the more restrictive type of pause. */
+ * In such a case, the duration is set to the new time and the new type is
+ * set to the more restrictive type of pause. */
 void pauseClients(mstime_t end, pause_type type) {
     if (type > server.client_pause_type) {
         server.client_pause_type = type;
     }
 
-    if (end > server.client_pause_end_time) {
-        server.client_pause_end_time = end;
-    }
+    server.client_pause_end_time = end;
```

Comment:
It seems like the real bug is that server.client_pause_end_time is not reset properly on failover, which I believe was fixed in here: https://github.com/redis/redis/commit/f560531d5b8a6e6d810b62114e69a5ffda7730f7. 

From a behavior perspective, the intended behavior of this check is that if there are two systems requesting a pause, the ordering shouldn't really matter and we should default to blocking until the latter is complete.

tl;dr, I don't think we should take the change, but we should keep the tests and make sure the bug is really fixed.

Reply:
I think we should move this line into the `if` section, means reset end time only when pause type changed.

Reply:
If we're really worried about that level of detail, I would prefer keeping track of the end times independently. If you go dig through the original PR, I was tracking both of them and a CLIENT PAUSE ALL didn't change the timing of CLIENT PAUSE RW, it was just a lot of code and I decided to drop it. Since, we generally want people to start using CLIENT PAUSE RW instead, since it's better and safer.

Reply:
@madolson  @soloestoy 
Thanks for your reply~

> From a behavior perspective, the intended behavior of this check is that if there are two systems requesting a pause, the ordering shouldn't really matter and we should default to blocking until the latter is complete.

I agree with this idea 😊 and i will go dig through the original PR.

Thanks for your comments.

Reply:
@bugwz you still working on this?